src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -87,9 +91,23 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
                     }
                     break;
                 case RANDOM:
-                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) >> 16);
+                    // start with a random number between Short.MIN_VALUE/4 and Short.MAX_VALUE/4
+                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) & 0b11_1111_1111_1111);
+                    // subtract Short.MIN_VALUE/4 from it to end up with a number between
+                    // 0 and Short.MAX_VALUE/2. This number can be then discounted in case the read is
+                    // not passing filters. We need to stay far from overflow so that when we add the two
+                    // scores from a read-pair we do not overflow since that could cause us to chose the
+                    // Failing read-pair instead of a passing one.
+                    score -= Short.MIN_VALUE/4;
             }
 
+            // make sure that filter-failing records are heavily discounted. Dividing by 2 to be far from
```

Comment:
This comment really applies to lines 109, but there is the handling of overflow and a comment regarding that between this comment and line 109, which is confusing.  Can you move this comment down to immediately before line 109?